### PR TITLE
Enabling Circle Ci integration

### PR DESCRIPTION
**Pull request by Darkvar**
Adding `config.yml` for Circle CI integration
Ticket: https://issues.sonatype.org/browse/NEXUS-21033

### CODE REVIEW

**Comment by Darkvar** [comment]
Travis is failing since #52 is not merged yet.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+mvn-deploy-command: &mvn-deploy-command
+  mvn-deploy-command: mvn -s .maven.xml clean deploy -PbuildKar -DdeployAtEnd=true -DperformRelease=true -DskipTests -Dspotbugs.skip=true
+  context: rso-base
+
+mvn-collect-artifacts-command: &mvn-collect-artifacts-command
+  mvn-collect-artifacts-command: |
+    mkdir -p ~/project/artifacts/junit/
+    find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} ~/project/artifacts/junit/ \;
+#   Uncomment this when adding
+#   cp ~/project/nexus-repository-r-it/target/failsafe-reports/*.xml ~/project/artifacts/junit/
```
Comment (bhamail) [inline .circleci/config.yml]: minor: Missing suffix in comment?, e.g. ...when adding 'integration tests'.

**Comment by Darkvar** [comment]
@DarthHater please take 2nd look. TravisCI config was removed, CircleCI badge was added (it will be empty till the job is enabled).

**Comment by Darkvar** [comment]
@DarthHater this PR is waiting for your review

**Comment by bhamail** [comment]
Very minor optional request: If you have a chance, could you move the `.maven.xml` file into the `.circleci` folder, and point the "deploy" target there as well?

See this commit for an example: https://github.com/sonatype-nexus-community/nexus-repository-conda/commit/71434aacd0de175b6a046cb6eef4c74b575f1b50

My +1 still holds with or without this change.

**Comment by Darkvar** [comment]
> Very minor optional request: If you have a chance, could you move the `.maven.xml` file into the `.circleci` folder, and point the "deploy" target there as well?

@bhamail I'd prefer to make such change in the separate PR. This PR doesn't contain `maven.xml` so it's not clear for me.

**Pull request merged**